tests/tests/lib_tests.rs

```diff
@@ -588,10 +590,132 @@ test_contract!(
 
 test_contract!(
     test_stx_transfer_err4,
-    "tokens",
+    "stx-funcs",
     "test-stx-transfer-err4",
     |response: ResponseData| {
         assert!(!response.committed);
         assert_eq!(*response.data, Value::UInt(4));
     }
 );
+
+#[test]
+fn test_define_ft() {
+    let contract_id = QualifiedContractIdentifier::new(
+        StandardPrincipalData::transient(),
+        ContractName::from("define-tokens"),
+    );
+    let mut datastore = Datastore::new();
+    let constants = StacksConstants::default();
+    let burn_datastore = BurnDatastore::new(constants);
+    let mut clarity_store = MemoryBackingStore::new();
+    let mut conn = ClarityDatabase::new(&mut datastore, &burn_datastore, &burn_datastore);
+    conn.begin();
+    conn.set_clarity_epoch_version(StacksEpochId::latest());
+    conn.commit();
+    let cost_tracker = LimitedCostTracker::new_free();
+    let mut contract_context = ContractContext::new(contract_id.clone(), ClarityVersion::latest());
+
+    let contract_str = std::fs::read_to_string("contracts/define-tokens.clar").unwrap();
+    let mut compile_result = compile(
+        contract_str.as_str(),
+        &contract_id,
+        cost_tracker,
+        ClarityVersion::latest(),
+        StacksEpochId::latest(),
+        &mut clarity_store,
+    )
+    .expect("Failed to compile contract.");
+
+    contract_context.set_wasm_module(compile_result.module.emit_wasm());
+
+    let mut global_context = GlobalContext::new(
+        false,
+        CHAIN_ID_TESTNET,
+        conn,
+        compile_result.contract_analysis.cost_track.take().unwrap(),
+        StacksEpochId::latest(),
+    );
+    global_context.begin();
+
+    {
+        initialize_contract(
+            &mut global_context,
+            &mut contract_context,
+            None,
+            &compile_result.contract_analysis,
+        )
+        .expect("Failed to initialize contract.");
+
+        let ft_metadata = contract_context
+            .meta_ft
+            .get("foo")
+            .expect("FT 'foo' not found");
+        assert_eq!(ft_metadata.total_supply, None);
+
+        let ft_metadata = contract_context
+            .meta_ft
+            .get("bar")
+            .expect("FT 'bar' not found");
+        assert_eq!(ft_metadata.total_supply, Some(1000000u128));
+    }
```

Comment:
I'm a little bit annoyed that this part that is the only one to change compared to the "macro-one" makes us write an entire test for it, and the `test_define_nft` function.
Shouldn't we instead add a new rule to `test_contract` that takes a `tt` and where we give just this part?

If those would be the only definitions that requires this writing, no problem. But otherwise I believe we can just make our macro more generic. (This could also be done in another PR, I will no block this one just for this detail)

Reply:
@obycode don't hesitate to resolve this conversation yourself if you find it irrelevant.

Reply:
It's those checks, plus the fact that we don't call any function, just initialize. So would it then make sense to have a separate macro for initialize-only tests?

Reply:
And the function call macro can first call the initialize macro, so that we don't duplicate anything?

Reply:
yes sure, but the actual macro could use the new rule.
But once again, not needed for this PR. If you want, I can do it later.

Reply:
Added in b20e7ac

Reply:
I did not think we needed an entire new macro for that, just a new rule in the existing one. But thank you for that :)